pkg/platform/aks/aks_test.go

```diff
@@ -0,0 +1,315 @@
+package aks
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclparse"
+
+	lokoconfig "github.com/kinvolk/lokomotive/pkg/config"
+)
+
+const (
+	testWorkerCount = 1
+)
+
+// createTerraformConfigFile()
+func TestCreateTerraformConfigFile(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "lokoctl-tests-")
+	if err != nil {
+		t.Fatalf("creating tmp dir should succeed, got: %v", err)
+	}
+
+	defer func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			t.Logf("failed to remove temp dir %q: %v", tmpDir, err)
+		}
+	}()
+
+	c := &config{
+		WorkerPools: []workerPool{
+			{
+				Name:   "foo",
+				VMSize: "bar",
+				Count:  testWorkerCount,
+			},
+		},
+	}
+
+	if err := createTerraformConfigFile(c, tmpDir); err != nil {
+		t.Fatalf("creating Terraform config files should succeed, got: %v", err)
+	}
+}
+
+func TestCreateTerraformConfigFileNoWorkerPools(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "lokoctl-tests-")
+	if err != nil {
+		t.Fatalf("creating tmp dir should succeed, got: %v", err)
+	}
+
+	defer func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			t.Logf("failed to remove temp dir %q: %v", tmpDir, err)
+		}
+	}()
+
+	c := &config{}
+
+	if err := createTerraformConfigFile(c, tmpDir); err == nil {
+		t.Fatalf("creating Terraform config files should fail if there is no worker pools defined")
+	}
+}
+
+func TestCreateTerraformConfigFileNonExistingPath(t *testing.T) {
+	c := &config{}
+
+	if err := createTerraformConfigFile(c, "/nonexisting"); err == nil {
+		t.Fatalf("creating Terraform config files in non-existing path should fail")
+	}
+}
+
+// Meta()
+func TestMeta(t *testing.T) {
+	assetDir := "foo"
+
+	moreWorkers := 3
+
+	c := &config{
+		AssetDir: assetDir,
+		WorkerPools: []workerPool{
+			{
+				Count: testWorkerCount,
+			},
+			{
+				Count: moreWorkers,
+			},
+		},
+	}
+
+	expectedNodes := 4
+	if e := c.Meta().ExpectedNodes; e != expectedNodes {
+		t.Errorf("Meta should count workers from all pools. Expected %d, got %d", expectedNodes, e)
+	}
+
+	if a := c.Meta().AssetDir; a != assetDir {
+		t.Errorf("Meta should return configured asset dir. Expected %q, got %q", assetDir, a)
+	}
+}
+
+// checkWorkerPoolNamesUnique()
+func TestCheckWorkerPoolNamesUniqueDuplicated(t *testing.T) {
+	c := &config{
+		WorkerPools: []workerPool{
+			{
+				Name: "foo",
+			},
+			{
+				Name: "foo",
+			},
+		},
+	}
+
+	if d := c.checkWorkerPoolNamesUnique(); !d.HasErrors() {
+		t.Fatalf("should return error when worker pools are duplicated")
+	}
+}
+
+func TestCheckWorkerPoolNamesUnique(t *testing.T) {
```

Comment:
I've written a few hours ago very similar tests [here](https://github.com/kinvolk/lokomotive/pull/119/commits/a335b6568de6bcb190ff6c718d2c0059b561154d#diff-52eb06f4b1a0db223eb4903d92c60c60R24-R61)

Writing it once seems fine, as we will probably re-use the code. But... more than once seems like a waste (because we are not doing c&p either!). Sorry I haven't looked at this before :-/